js/test/date/testdatefmt_eu_ES.js

```diff
@@ -0,0 +1,1788 @@
+/*
+ * testdatefmt_eu_ES.js - test the date formatter object in Basque/Spain
+ * 
+ * Copyright © 2020, JEDLSoft
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+if (typeof(JulianDate) === "undefined") {
+    var JulianDate = require("../../lib/JulianDate.js");
+}
+if (typeof(GregorianDate) === "undefined") {
+    var GregorianDate = require("../../lib/GregorianDate.js");
+}
+if (typeof(DateFmt) === "undefined") {
+    var DateFmt = require("../../lib/DateFmt.js");
+}
+if (typeof(ilib) === "undefined") {
+    var ilib = require("../../lib/ilib.js");
+}
+
+module.exports.testdatefmt_eu_ES = {
+    setUp: function(callback) {
+        ilib.clearCache();
+        callback();
+    },
+
+    testDateFmtConstructorEmpty_eu_ES: function(test) {
+        test.expect(1);
+        var fmt = new DateFmt({locale: "eu-ES"});
+        
+        test.ok(fmt !== null);
+        test.done();
+    },
+
+    testDateFmtSimpleShort_eu_ES: function(test) {
+        test.expect(2);
+        var fmt = new DateFmt({locale: "eu-ES", length: "short"});
+        test.ok(fmt !== null);
+        
+        var date = new GregorianDate({
+            locale: "eu-ES",
+            year: 2011,
+            month: 9,
+            day: 29,
+            hour: 13,
+            minute: 45,
+            second: 0,
+            millisecond: 0
+        });
+        test.equal(fmt.format(date), "11/9/29");
+        test.done();
+    },
+
+    testDateFmtSimpleMedium_eu_ES: function(test) {
+        test.expect(2);
+        var fmt = new DateFmt({locale: "eu-ES", length: "medium"});
+        test.ok(fmt !== null);
+        
+        var date = new GregorianDate({
+            locale: "eu-ES",
+            year: 2011,
+            month: 9,
+            day: 29,
+            hour: 13,
+            minute: 45,
+            second: 0,
+            millisecond: 0
+        });
+        test.equal(fmt.format(date), "2011(e)ko ira. 29(a)");
```

Comment:
From perusing various basque web sites, it seems that the (e) and the (a) are optional and are used when formatting the correct gender/case of the number. Sort like when you write "word(s)" in English to indicate a possible plural. That means they are optional parts that depend on the number that is being formatted. I am not sure this date format is acceptable to Basque people, or if it looks really awkward. Please have this verified by any Basque LG members you can find.

Reply:
Regarding `eu-ES` locale, it's not officially required to support at this locale now, and it's just preparing for reference platform perspective. so It's hard to find(or contact) Basque people. The data is directly from CLDR without modifying it. I think It's better to ask to CLDR.

Reply:
Different PRs are better. It is easier to see the differences, and merging the one file is not so difficult.

Reply:
ok. I'll do that way.